### PR TITLE
[FIX] improved export of consensusXML

### PR DESCRIPTION
**Pull request by timosachsenberg**
ConsensusXML export is fixed and much more conform to mzTab standard.
Meta values attached to consensus feature identifications and hits are exported in additional columns.
If multiple identifications are attached an error is given and the user instructed to run IDConflictResolver first.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
any objections on the merge? travis passed and export is more correct than before...

File: src/topp/MzTabExporter.cpp [hunk 1]
```diff
@@ -650,6 +650,23 @@ namespace OpenMS
 //        MzTabString db = sp.db.empty() ? MzTabString() : MzTabString(sp.db);
 //        MzTabString db_version = sp.db_version.empty() ? MzTabString() : MzTabString(sp.db_version);
 
+        // currently we don't save how many channels we saved in a consensus feature so we scan the file
+        // for the maximum max index in a consensus feature and take this as number of study variables
```
Comment (cbielow) [inline src/topp/MzTabExporter.cpp]: mhh.. how about 
const FileDescriptions & 	getFileDescriptions () const
This should get you what you want, faster and also coverting files which have no featurehandle (unlikely to happen though... they are empty then I'd think).

File: src/topp/MzTabExporter.cpp [hunk 2]
```diff
@@ -686,25 +755,115 @@ namespace OpenMS
...
+          vector<PeptideIdentification> pep_ids = c.getPeptideIdentifications();
+          if (!pep_ids.empty())
+          {
+            if (pep_ids.size() != 1)
+            {
+              throw OpenMS::Exception::IllegalArgument(__FILE__, __LINE__, __FUNCTION__, "Consensus features might contain at most one identification. Run IDConflictResolver first to remove ambiguities!");
+            }
+
+            pep_ids[0].assignRanks();
+            const PeptideHit& best_ph = pep_ids[0].getHits()[0];
```
Comment (cbielow) [inline src/topp/MzTabExporter.cpp]: is it guaruanteed that there is at least one hit?

**Comment by cbielow** [comment]
could we have TOPP test as well?!
Especially the new opt columns could be a bit tricky to fill/align.

**Pull request merged**